mono/metadata/marshal-ilgen.c

```diff
@@ -1370,9 +1370,19 @@ emit_invoke_call (MonoMethodBuilder *mb, MonoMethod *method,
 	/* to make it work with our special string constructors */
 	if (!string_dummy) {
 		ERROR_DECL (error);
+
+		// FIXME Allow for static construction of MonoString.
+
+		SETUP_ICALL_FUNCTION;
+		SETUP_ICALL_FRAME;
+
 		MONO_GC_REGISTER_ROOT_SINGLE (string_dummy, MONO_ROOT_SOURCE_MARSHAL, NULL, "Marshal Dummy String");
-		string_dummy = mono_string_new_checked (mono_get_root_domain (), "dummy", error);
```

Comment:
Since this storing right into a registered global, it is ok, but I'd like to remove all calls to `mono_string_new_checked`. `MONO_HANDLE_RAW` we get in trade, and it isn't clearly a good trade.
Perhaps we need some special things like `mono_foo_new_global (&global, …)`.

Reply:
Perhaps `MONO_GC_REGISTER_ROOT_SINGLE` should be augmented.